src/dispensary.js

```diff
@@ -0,0 +1,218 @@
+import fs from 'fs';
+
+import async from 'async';
+import request from 'request';
+
+import { DEFAULT_HAHES_FILE, DEFAULT_LIBRARY_FILE } from 'const';
+import hasher from 'hasher';
+import log from 'logger';
+import { getVersions } from 'versions';
+
+
+function fileFormat(url, {file=null, version=null}={}) {
+  if (!file || !version) {
+    throw new Error('ArgumentError: File and version are required.');
+  }
+
+  return url.replace('$VERSION', version)
+            .replace('$FILENAME', file)
+            .replace('$VERSION', version)
+            .replace('$FILENAME', file);
+}
+
+export default class Dispensary {
+
+  constructor(config={}, _libraries=null, _hashes=DEFAULT_HAHES_FILE) {
+    this._libraries = _libraries;
+    this.libraryFile = DEFAULT_LIBRARY_FILE;
+    this.hashesFile = _hashes;
+
+    if (config._ && config._[0]) {
```

Comment:
This `config._` syntax is a bit strange. What does `_` represent? I see it being used in the tests.